src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3326,13 +3328,15 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
 		}
 	}
 
-	si_tcp_logdbg("going to OS for setsockopt level %d optname %d", __level, __optname);
-	ret = orig_os_api.setsockopt(m_fd, __level, __optname, __optval, __optlen);
-	BULLSEYE_EXCLUDE_BLOCK_START
-	if (ret) {
-		si_tcp_logdbg("setsockopt failed (ret=%d %m)", ret);
+	if (sys_call) {
```

Comment:
If the user is root he should not skip this system call regardless of "avoid_sys_call" parameter. Please update logic. 
Can you think of a way to prevent errors if you do not have root priviledges and parameter avoid_sys_call is true?